lib/util/SortableSet.js

```diff
@@ -0,0 +1,28 @@
+"use strict";
+
+module.exports = class SortableSet extends Set {
+
+	constructor(initialIterable, defaultSort) {
+		super(initialIterable);
+		this._sortFn = defaultSort;
+	}
+
+	/**
+	 * @param {Function} sortFn - function to sort the set
+	 * @returns {void}
+	 */
+	sortWith(sortFn) {
+		const sortedArray = Array.from(this).sort(sortFn);
```

Comment:
The original impl checked if the set was already sorted before sorting. The new SortableSet should do the same. `add` and `clear` should be overriden to invalidate sorting. Best store the current active sortFn.